src/js/controller/controller.js

```diff
@@ -1036,14 +1053,21 @@ Object.assign(Controller.prototype, {
         };
 
         this.setPlaylistItem = function (index, item) {
-            item = normalizePlaylistItem(_model, new Item(item), item.feedData || {});
-
-            if (item) {
-                const playlist = _model.get('playlist');
-                playlist[index] = item;
-
-                if (index === _model.get('item') && _model.get('state') === 'idle') {
-                    this.setItemIndex(index);
+            const playlist = _model.get('playlist');
+            const playlistItem = playlist[index];
+            if (item && item !== playlistItem) {
+                const newItem = normalizePlaylistItem(_model, new Item(item), item.feedData || {});
+                if (newItem) {
```

Comment:
Correct me if I am wrong, but shouldn't we be replacing the playlistItem with this `newItem` somewhere?
`playlist[index] = newItem` is missing